software/firmware/europi.py

```diff
@@ -209,15 +209,20 @@ def __init__(self, pin, min_voltage=MIN_INPUT_VOLTAGE, max_voltage=MAX_INPUT_VOL
                 )
         self._gradients.append(self._gradients[-1])
 
-    def percent(self, samples=None):
+    def percent(self, samples=None, deadzone=None):

```

Comment:
[discussion] I'm wondering of we should take the deadzone parameter, but ignore it (and document this behavior).

I believe that we've disscused before that deadzone makes sense for knobs, but less so for ain. I'm not sure that we should reverse that decision hastily to fix a bug.

Reply:
yes, actually it was me that brought up the point that deadzones make no sense for ain.
And I had a different solution to fix this issue first, where ain.choice would just not have a deadzone parameter.
However I decided that it is a nicer solution if k1.choice and ain.choice still have the same usage.
Sure, taking and ignoring the parameter is an option, but that feels wrong as well...
As mentioned the deadzones will be set to 0 for a in anyway unless the user overwrites that.
And maybe there are some cases where deadzones on ain are actually useful (especially with range and choice)?
If speed is your concern we could place a 'if deadzones!=0.0' to skip parts of the code

Reply:
Here is an alternative PR: https://github.com/Allen-Synthesis/EuroPi/pull/253